app/pages/project/instances/instance/InstancePage.tsx

```diff
@@ -99,6 +101,11 @@ export function InstancePage() {
     query: { project: instanceSelector.project },
   })
 
+  useInterval({
+    fn: () => apiQueryClient.invalidateQueries('instanceView'),
+    delay: instanceTransitioning(instance) ? 1000 : null,
+  })
+
```

Comment:
We're doing it this way instead of `refetchInterval` on the `useApiQuery` call because of this: if that call is what defines the instance, it can't also rely on the instance. We can probably hack around that by making two calls, passing the `instance` from the first to the second, but that is somewhat appalling. `useInterval` is more flexible probably. For example, it makes it easy to use the `refreshData` function as the callback if we want. I kind of want to avoid that because it makes 5 calls instead of one.

![image](https://github.com/user-attachments/assets/6f228b88-2c26-42e4-bad7-18cdf0eff0b7)

Reply:
No longer the case! I figured out how to do look at the instance data in a `refetchInterval` callback. It was hidden in `query.state`.